src/Modules/Backend/Backend/Ajax/AjaxActionGetTwoFactorAuthorizationCode.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace ForkCMS\Modules\Backend\Backend\Ajax;
+
+use Endroid\QrCode\Builder\Builder;
+use Endroid\QrCode\Encoding\Encoding;
+use Endroid\QrCode\ErrorCorrectionLevel\ErrorCorrectionLevelHigh;
+use Endroid\QrCode\RoundBlockSizeMode\RoundBlockSizeModeMargin;
+use Endroid\QrCode\Writer\SvgWriter;
+use ForkCMS\Modules\Backend\Domain\AjaxAction\AbstractAjaxActionController;
+use ForkCMS\Modules\Backend\Domain\User\User;
+use Scheb\TwoFactorBundle\Security\TwoFactor\Provider\Google\GoogleAuthenticatorInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class AjaxActionGetTwoFactorAuthorizationCode extends AbstractAjaxActionController
+{
+    public function __construct(
+        private readonly GoogleAuthenticatorInterface $googleAuthenticator,
+        private readonly TokenStorageInterface $tokenStorage
+    ) {
+    }
+
+    public function execute(Request $request): void
+    {
+        $user = $this->tokenStorage->getToken()->getUser();
+
+        if (!$user instanceof User) {
+            $this->assign('error', 'User not found');
+
+            return;
+        }
+
+        $secret = $this->googleAuthenticator->generateSecret();
+        $user->setGoogleAuthenticatorSecret($secret);
+        $code = $this->googleAuthenticator->getQRContent($user);
+        $this->assign('secret', $secret);
+        $this->assign('code', $code);
+
+        // @TODO: Make more generic
```

Comment:
Is this todo for a different PR or did you forget it?

Reply:
This is a todo for later down the line, @tijsverkoyen requested to make it more generic for global use in https://github.com/forkcms/forkcms/pull/3567